src/auth/user.rs

```diff
@@ -16,6 +19,14 @@ pub trait UserDetail: Send + Sync + Display + Debug {
     fn account_enabled(&self) -> bool {
         true
     }
+
+    /// Returns the user's home directory, if any.  If the user has a home directory, then their
+    /// sessions will be restricted to this directory.
+    ///
+    /// The path should be absolute.
+    fn home(&self) -> Option<&Path> {
```

Comment:
The problem with relative home directories is that it embeds information about the storage back-end into the user authentication engine.  I chose absolute paths so that, for example, one can configure unftp-auth-jsonfile for multiple users, and then configure unftp-sbe-fs separately.  Neither needs to know about the other's configuration.

Reply:
> embeds information about the storage back-end into the user authentication engine

We circumvent this in unFTP by using the delegate pattern: See [here](https://github.com/bolcom/unFTP/blob/56d9c7f226fb7e5bbf3a8ccaa6824ecaae84f03e/src/auth.rs#L74C2-L74C2). 

Like mentioned the libunftp interfaces here probably needs a rethink.

Reply:
So the authenticator has a way to ask the storage backend what its root directory is?

Reply:
no the storage back-end receives an instance of the UserDetail implementation and it can then use the directory defined there and base it relative to the storage back-end root. In fact you probably want to change the enter method to receive a UserDetail implementation instead of a path.